addons/mail/models/discuss/discuss_channel.py

```diff
@@ -102,15 +102,11 @@ def _constraint_group_id_channel(self):
 
     # COMPUTE / INVERSE
 
-    @api.depends('channel_type', 'is_member')
+    @api.depends('channel_type', 'is_member', 'group_public_id')
+    @api.depends_context('uid')
     def _compute_is_editable(self):
         for channel in self:
-            if channel.channel_type == 'channel':
-                channel.is_editable = self.env.user._is_admin() or channel.create_uid.id == self.env.user.id
-            elif channel.channel_type == 'group':
-                channel.is_editable = channel.is_member and not self.env.user.share
-            else:
-                channel.is_editable = False
+            channel.is_editable = channel.check_access_rights('write', raise_exception=False)
```

Comment:
You always need to check both check_access_rights and check_access_rule at the same time (csv and xml).